gateway/metrics/exporter.go

```diff
@@ -57,11 +94,7 @@ func AttachExternalWatcher(endpointURL url.URL, metricsOptions MetricOptions, la
 					continue
 				}
 
-				for _, service := range services {
-					metricsOptions.ServiceReplicasCounter.
-						WithLabelValues(service.Name).
-						Set(float64(service.Replicas))
-				}
+				e.services = services
```

Comment:
I've decided not to change much of the current implementation. So this timer is the same as before but just caches the services. When prometheus scrapes the metrics, this cache is used to provide the replica counts.